packages/loader/container-loader/src/container.ts

```diff
@@ -1036,6 +1044,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             args.mode = "write";
         }
 
+        // Notify listeners that we are attempting to resume the delta connection
+        this.emit("resumeConnection", args);
```

Comment:
I do not think there is guarantee that we get here when disconnected. For example, Container.resume() can be called any number of times in a row.

Reply:
Hmmm great point. I think just checking connection state somehow for disconnected should be ok.

If someone calls resume twice in a row, and the first one hasn't yielded a connection yet, I think it's fair to emit this event twice. The caller would be listening for the connected event, and would be able to easily dedupe them. But I agree that once connected, another call to resume should not result in this event.

Reply:
Updated it - and this should probably move to be managed by connectionStateHandler, maybe along with #7884